libs/agno/agno/workflow/parallel.py

```diff
@@ -205,19 +207,30 @@ def execute(
 
         self._prepare_steps()
 
+        # Create individual session_state copies for each step to prevent race conditions
+        session_state_copies = []
+        for _ in range(len(self.steps)):
+            if session_state is not None:
+                session_state_copies.append(deepcopy(session_state))
+            else:
+                session_state_copies.append({})
+
         def execute_step_with_index(step_with_index):
             """Execute a single step and preserve its original index"""
             idx, step = step_with_index
             try:
+                # Use the individual session_state copy for this step
+                step_session_state = session_state_copies[idx]
```

Comment:
rather extract before the try and reference this instance in the `except` as well